target/linux/ath79/image/generic-tp-link.mk

```diff
@@ -0,0 +1,30 @@
+include ./common-tp-link.mk
+
+define Device/tl_wr1043nd_v1
+  $(Device/tplink-8m)
+  ATH_SOC := ar9132
+  DEVICE_TITLE := TP-LINK TL-WR1043N/ND v1
+  DEVICE_PACKAGES := kmod-usb-core kmod-usb2 kmod-usb-ledtrig-usbport
+  BOARDNAME := TL-WR1043ND
+  DEVICE_PROFILE := TLWR1043
+  TPLINK_HWID := 0x10430001
+endef
+TARGET_DEVICES += tl_wr1043nd_v1
+
+define Device/tl-wr1043nd-v2
+  $(Device/tplink-8mlzma)
+  ATH_SOC := qca9558
+  DEVICE_TITLE := TP-LINK TL-WR1043N/ND v2
+  DEVICE_PACKAGES := kmod-usb-core kmod-usb2 kmod-usb-ledtrig-usbport
+  BOARDNAME := TL-WR1043ND-v2
+  DEVICE_PROFILE := TLWR1043
+  TPLINK_HWID := 0x10430002
+endef
+TARGET_DEVICES += tl-wr1043nd-v2
+
+define Device/tl-wr1043nd-v3
+  $(Device/tl-wr1043nd-v2)
```

Comment:
same as with the dtsi/dts files. Move common settings to a common "Device/tl-wr1043nd". Don't overwrite settings made in the recipe you include (see #894).

It is really hard to read and might has unintended side effects if Device/tl-wr1043nd-v2 need to be changed one time.

Here it don't see much advantage in creating a common recipe. The only common settings are ATH_SOC and DEVICE_PACKAGES (BOARDNAME and DEVICE_PROFILE isn't really required and already dropped in my staging tree).